src/DynamicNLPModels.jl

```diff
@@ -257,18 +297,28 @@ function LQDynamicModel(dnlp::LQDynamicData{T,S,M}; condense = false) where {T,S
         end
 
         d    = fill(Inf, ns * 2)
-        Jcon = zeros(ns * 2, ns)
-        E    = zeros(ns * 2, nu)
 
-        for i in 1:ns
-            Jcon[i,i] = -1.0
-            Jcon[i + ns, i] = 1.0
-        end
-        
         d[1:ns] .= .-sl
         d[(ns+1):(2*ns)] .= su
 
-        J, ucon = _build_G(block_A, block_B, Jcon, E, d, s0, N)
+
+        E_bounds = zeros(ns * 2, ns)
+        F_bounds = zeros(ns * 2, nu)
+
+        for i in 1:ns
+            E_bounds[i,i] = -1.0
+            E_bounds[i + ns, i] = 1.0
+        end
+
+        E_bounds = vcat(E_bounds, E)
+        F_bounds = vcat(F_bounds, F)
+        d        = vcat(d, gu)
+
```

Comment:
Double-sided inequalities should be used; in this way, the NLP solver needs to handle x2 more constraints